xoverturning/compfunc.py

```diff
@@ -165,8 +172,24 @@ def select_basins(
     elif basin == "indopac":
         maskbin = ds[mask].where((basincodes == 3) | (basincodes == 5))
 
-    maskmoc = xr.where(maskbin == 1, True, False)
-    return maskmoc
+    maskbasin = xr.where(maskbin == 1, True, False)
+
+    # create a mask for the streamfunction
+    if bathy in ds:
+        if "yq" in maskbasin.dims:
+            grid = define_grid(ds)
+            bathy_coloc = grid.interp(ds[bathy], "Y", boundary="fill")
+        elif "yh" in maskbasin.dims:
+            bathy_coloc = ds[bathy]
+        else:
+            raise ValueError("Unsupported coord")
+        bathy_basin = bathy_coloc.where(maskbasin).fillna(0.0)
+        max_depth = bathy_basin.max(dim="xh")
+        maskmoc = xr.where(ds["z_i"] > max_depth, 0, 1)
```

Comment:
"z_i" will not be present when using the "rho_2" vertical coord. and there is no check here to determine the coordinate system

Reply:
yes, I have a bit of cleanup to do